GMX-Bots/gmx5-detect-frontruns/src/agent.ts

```diff
@@ -0,0 +1,206 @@
+        if (callHistory.has(accountBack + token0 + token1)) {
+          let potentialSandwhichFront: [LogDescription, number, string] = callHistory.get(
+            accountBack + token0 + token1
+          )!;
+          const {
+            account: accountFront,
+            tokenIn: tokenInFront,
+            tokenOut: tokenOutFront,
+            amountIn: amountInFront,
+            amountOut: amountOutFront,
+          } = potentialSandwhichFront[0].args;
+
+          //check if the front transaction trades the same tokens in the opposite direction as the back transaction
+          if (tokenInBack == tokenOutFront && tokenOutBack == tokenInFront) {
+            //find any victim transaction
+            callHistoryKeys.forEach((addressAndTokens) => {
+              let event = callHistory.get(addressAndTokens)!;
+              const {
+                account: accountVictim,
+                tokenIn: tokenInVictim,
+                tokenOut: tokenOutVictim,
+                amountIn: amountInVictim,
+                amountOut: amountOutVictim,
+              } = event[0].args;
+              //check if victim was inside the sandwich
+              if (
+                (event[1] > potentialSandwhichFront[1] &&
+                  event[1] < callHistoryIndex &&
+                  callHistoryIndex > potentialSandwhichFront[1]) ||
+                (event[1] < potentialSandwhichFront[1] &&
+                  event[1] < callHistoryIndex &&
+                  callHistoryIndex < potentialSandwhichFront[1]) ||
+                (event[1] > potentialSandwhichFront[1] &&
+                  event[1] > callHistoryIndex &&
+                  callHistoryIndex < potentialSandwhichFront[1])
+              ) {
+                //check if the victim transaction trades the same tokens in the same direction as the front transaction
+                if (tokenInVictim == tokenInFront && tokenOutVictim == tokenOutFront) {
+                  sandwichAttack = true;
+                  victimAddress = accountVictim;
+                  frontHash = potentialSandwhichFront[2];
+                  victimHash = event[2];
+                  victimTokenIn = tokenInVictim;
+                  victimTokenOut = tokenOutVictim;
+                  victimTokenInAmount = amountInVictim;
+                  victimTokenOutAmount = amountOutVictim;
+                  frontrunnerAddress = accountFront;
+                  frontrunnerProfit = amountOutBack - amountInFront;
+                  victimAddressAndTokens = addressAndTokens;
+                  frontrunnerAddressAndTokens = accountBack + token0 + token1;
+                }
+              }
+            });
+          }
+        }
+
+        if (!sandwichAttack) {
+          callHistory.set(accountBack + token0 + token1, [swapEvent, callHistoryIndex, txEvent.hash]);
+          callHistoryKeys[callHistoryIndex] = accountBack + token0 + token1;
+        }
+
+        callHistoryIndex++;
+        mapCleanerCounter++;
```

Comment:
Why the Index is incremented even if a sandwichAttack is detected?

Reply:
Causes no real problems but should not be incremented in this situation, fixed it